.github/scripts/update_version.py

```
@@ -0,0 +1,108 @@
+    """Extract version string from file content."""
+    match = re.search(r"version = ['\"]([^'\"]+)['\"]", content)
+    if not match:
+        raise ValueError("Could not find version string in file")
+    return match.group(1)
+
+
+def parse_calver(version: str) -> tuple[int, int, int]:
+    """Parse CalVer string into (year, month, micro) components."""
+    parts = version.split('.')
+    if len(parts) != 3:
+        raise ValueError(f"Invalid CalVer format: {version}")
```

Ruff (0.14.5)

Avoid specifying long messages outside the exception class

(TRY003)